src/components/BalanceInfo/BalanceInfo.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import ReferenceStore from '../../stores/referenceStore';
+import Select from '../Select/select';
+import './balance-info.css';
+import {BalanceInfoProps} from './index';
+
+const StyledSelect = styled(Select)`
+  .Select-control {
+    background-color: #333333 !important;
+
+    .Select-value {
+      .Select-value-label {
+        color: #fff !important;
+      }
+    }
+
+    .Select-input {
+      input {
+        color: #fff;
+      }
+      div {
+        background-color: #333333 !important;
+        color: #fff !important;
+      }
+    }
+  }
+
+  .Select-menu-outer {
+    .Select-option,
+    .Select-noresults {
+      background-color: #333333 !important;
+      color: #fff !important;
+
+      &.is-focused,
+      &.is-selected {
+        background-color: #666666 !important;
+      }
+    }
+  }
+`;
+
+const StyledButton = styled.button`
+  background: none;
+  outline: none;
+  border: none;
+  color: #fff;
+
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+class BalanceInfo extends React.Component<BalanceInfoProps> {
+  private readonly referenceStore: ReferenceStore = this.props.referenceStore;
+  private readonly uiStore = this.props.uiStore;
+
+  documentClickHandler = (e: any) => {
+    const target: any = e.target;
+    if (target.closest('.Select') || target.id === 'baseAssetBtn') {
+      return;
+    }
+    this.uiStore.toggleAssetsSelect();
+    document.body.removeEventListener('click', this.documentClickHandler);
+  };
+
+  onClickHandler = () => {
+    this.uiStore.toggleAssetsSelect();
+    document.body.addEventListener('click', this.documentClickHandler);
```

Comment:
`componentDidMount` and `componentWillUnmount` are better places for attaching/detaching handlers